packages/react/mixin.server.js

```diff
@@ -56,28 +53,25 @@ class ReactPlugin extends Mixin {
       .then((element) =>
         this.fetchData({}, element).then((data) => ({ element, data }))
       )
-      .then((result) => {
-        const { element, data } = result;
+      .then(({ element, data }) => {
         const markup = renderToString(element);
         const helmet = Helmet.renderStatic();
-        this.enhanceData(Object.assign({}, data, { markup, helmet }))
-          .then(template)
-          .then((document) => {
-            const routerContext = this.routerContext;
-            if (routerContext.miss) {
-              next();
-            } else if (routerContext.url) {
-              res.status(routerContext.status || 301);
-              res.set('Location', routerContext.url);
-              routerContext.headers && res.set(routerContext.headers);
-              res.end();
-            } else {
-              res.status(routerContext.status || 200);
-              routerContext.headers && res.set(routerContext.headers);
-              res.type('html');
-              res.send(document);
-            }
-          });
+        const { context = {} } = this.options;
```

Comment:
I kinda liked the name `routerContext` as it gave a clear differentiation to React's `context`

Reply:
Well, since we are talking about a local variable here (and ‘this.options’ is all about router options), I was thinking the naming was clear/self-documenting enough. It is called ‘context’ in the context of ‘ReactRouter’, anyway :)